### ISSUE TITLE
About Fine-tuning Strategy in `finetune.py`

### ISSUE THREAD

**Issue by noname31337** [lyhkevin/MT-Net#5]
Hello,

Firstly, great work on the project!

I've been reviewing the `finetune.py` script and noticed something about the fine-tuning strategy. In the paper, it's mentioned that you **"fine-tune the last six layers of the pre-trained encoder while freezing the others."** I might be missing something, but in the code, the pretrained encoder (`E`) is updating all of its layers:

```python
E = MAE_finetune(img_size=opt.img_size, patch_size=opt.mae_patch_size, embed_dim=opt.encoder_dim, depth=opt.depth,
                 num_heads=opt.num_heads, in_chans=1, mlp_ratio=opt.mlp_ratio)
```
I saw the gradients for `FC_module` are turned off:
```python
for param in FC_module.parameters():
    param.requires_grad = False
```

The optimizer is initialized to optimize over parameters of both `E` and `G`:
```python
params = list(E.parameters()) + list(G.parameters())
optimizer = torch.optim.Adam(params)
```

But I couldn't find any section in the code where the initial layers are frozen.

Could you please clarify if this fine-tuning strategy of only updating the last six layers is an essential step? I'm curious if I can reproduce your results without this specific strategy.

Thank you for your time and assistance!

**Comment by lyhkevin** [comment]
Hello! Fine-tuning with frozen initial layers is commonly employed when computational resources are limited. Nevertheless, if there is an abundant supply of paired data for fine-tuning (such as 100% paired data from the training set), this strategy can actually hamper the performance of the framework. Consequently, it is not a crucial step for enhancing the accuracy of the synthesis. However, it can prove useful in scenarios where the quantity of paired data is extremely limited, such as 10%, as it can help prevent the framework from overfitting.

**Comment by noname31337** [comment]
Thank you for the quick and clear explanation. I've also seen your mention of this in **Effects of Partial Fine-tuning**. Much appreciated.
